docs/backend/testing/running-integration-tests-with-routing-middleware-enabled.md

```diff
@@ -0,0 +1,105 @@
+---
+sidebar_position: 8
+---
+
+# Running integration tests with routing middleware enabled
+
+Sometimes it will be necessary to write server integration tests with routing middleware enabled. For example, we want to test calling an API endpoint with an unauthorized user, and ensure that we receive a 403 response. Here's how to go about it.
+
+## Test name and location
+
+Your test will live in `pkg/handlers/routing`, in a subdirectory corresponding to the API you're testing, `{api_name}api_test`. Tests of the `internal` API live in `pkg/handlers/routing/internalapi_test`.
+
+The name of the file should correspond to the handler that the endpoint calls, `{handler}_test.go`. `pkg/handlers/internalapi/ppm_shipment.go` would have a corresponding test file in `pkg/handlers/routing/internalapi_test/ppm_shipment_test.go`.
+
+Generally, the test name and location should be
+```pkg/handlers/routing/{api_name}api_test/{handler_name}_test.go```
```

Comment:
**suggestion (non-blocking)**: related to the suggestion I made above

```suggestion
Generally, the test name and location should be
```pkg/handlers/routing/{api_name}api_test/{handler_file_name}_test.go```
```